core/src/main/java/org/pentaho/platform/web/http/api/resources/GenericFileResource.java

```diff
@@ -196,10 +208,55 @@ public Response createFolder( @NonNull @PathParam( "path" ) String path ) {
     }
   }
 
+  @GET
+  @Path( "{path : .+}/content" )
+  @Produces( { MediaType.APPLICATION_JSON } )
+  @StatusCodes( {
+    @ResponseCode( code = 200, condition = "Operation successful" ),
+    @ResponseCode( code = 400, condition = "Filter is invalid" ),
+    @ResponseCode( code = 401, condition = "Authentication required" ),
+    @ResponseCode( code = 403, condition = "Access forbidden" ),
+    @ResponseCode( code = 500, condition = "Operation failed" )
+  } )
+  public Response getFileContent( @NonNull @PathParam( "path" ) String filePath )
+    throws OperationFailedException, IOException {
+
+    GenericFilePath genericFilePath = GenericFilePath.parseRequired( decodePath( filePath ) );
+    IGenericFileContentWrapper contentWrapper = genericFileService.getFileContentWrapper( genericFilePath );
+
+    return buildOkResponse( getStreamingOutput(contentWrapper.getInputStream()), contentWrapper.getFileName(), contentWrapper.getMimeType() );
```

Comment:
I'm sure you have a todo implement the ladder of catch throws just like the other methods to comply with the Response codes.

Reply:
this and documentation...